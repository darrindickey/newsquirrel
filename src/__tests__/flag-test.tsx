import 'jest-dom/extend-expect';
import * as React from 'react';
import { cleanup, render } from 'react-testing-library';
import Flag from '../components/flag/flag';

afterEach(cleanup);

test('Renders Flag component correctly', () => {
  const {getByTestId} = render(<Flag code="in" />);
  const node = getByTestId('rt-flag-wrapper');
  const icon = node.querySelector('i') as HTMLDivElement;

  expect(node).toBeInTheDocument();
  expect(icon).toBeVisible();
  expect(icon).toHaveAttribute('aria-label');
  expect(node).toMatchSnapshot();
});

