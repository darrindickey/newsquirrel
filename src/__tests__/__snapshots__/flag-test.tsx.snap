// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Flag component correctly 1`] = `
<div
  class="sc-bdVaJa cweobK"
  data-testid="rt-flag-wrapper"
>
  <i
    aria-label="icon"
    class="sc-bwzfXH hQhWXo"
  />
</div>
`;
